### DISCUSSION TITLE
Why does kernl not support gpus before ampere architecture?

### DISCUSSION THREAD

**Discussion by seastar105** [ELS-RD/kernl#276] [Q&A] [answered]
as @jonathlela said in #246, kernl doesn't support gpus before ampere(8.0), but in triton's readme, triton says they support compute capability 7.0+.

what's the reason kernl not supporting volta, turing architecture?

**Comment by jonathlela** [comment, marked as answer]
Currently, our attention kernels don't compile with triton on turing/volta, even the latest version. We didn't investigate much but it may be related to https://github.com/openai/triton/issues/616. We may be able to rewrite the forward pass of the kernel to make it works but we prefer to wait for a more stable release of triton before.
